src/games/strategy/triplea/util/TransportUtils.java

```diff
@@ -40,125 +35,70 @@
   }
 
   /**
-   * Returns a map of unit -> transport. Unit must already be loaded in the
-   * transport. If no units are loaded in the transports then an empty Map will
-   * be returned.
+   * Returns a map of unit -> transport. Tries to load units evenly across all transports.
    */
-  private static Map<Unit, Unit> mapTransportsAlreadyLoaded(final Collection<Unit> units,
+  public static Map<Unit, Unit> mapTransportsToLoad(final Collection<Unit> units,
       final Collection<Unit> transports) {
-    final Collection<Unit> canBeTransported = Match.getMatches(units, Matches.UnitCanBeTransported);
-    final Collection<Unit> canTransport = Match.getMatches(transports, Matches.UnitCanTransport);
+
+    final List<Unit> canBeTransported = sortByTransportCostDescending(units);
+    final List<Unit> canTransport = sortByTransportCapacityAscending(transports);
+
+    // Add units to transports evenly
     final Map<Unit, Unit> mapping = new HashMap<Unit, Unit>();
-    final Iterator<Unit> land = canBeTransported.iterator();
-    while (land.hasNext()) {
-      final Unit currentTransported = land.next();
-      final Unit transport = TransportTracker.transportedBy(currentTransported);
+    final IntegerMap<Unit> addedLoad = new IntegerMap<Unit>();
+    for (final Unit unit : canBeTransported) {
```

Comment:
Complexity looks to be getting up there in this method. Can we push some of the lower level logic into helper methods? Java8 streams + filter operations might be another option, the inner most for loop looks like it can be simplified to a stream, filter, find_first operation